Feature/integration tests

alexjoverm opened the pull request:
## Pull request type

Jira Link: [INT-](url)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Other (please describe):

Adding the setup for both E2E and integration tests + a basic test of each type, which would cover enough for now.

## How to test this PR

- Checkout this branch
- `npm install`
- `npm t` (this will run the new commands for integration and e2e tests

thiagosaife commented:
@alexjoverm my friend. All the unit tests ran as expected, but I'm getting an error when trying to initialize Cypress. I will put it down here:

> failed to load config from /Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/cypress/vite.config.js
> 
> You installed esbuild on another platform than the one you're currently using.
> This won't work because esbuild is written with native code and needs to
> install a platform-specific binary executable.
> 
> Specifically the "esbuild-darwin-arm64" package is present but this platform
> needs the "esbuild-darwin-64" package instead. People often get into this
> situation by installing esbuild on Windows or macOS and copying "node_modules"
> into a Docker image that runs Linux, or by copying "node_modules" between
> Windows and WSL environments.
> 
> If you are installing with npm, you can try not copying the "node_modules"
> directory when you copy the files over, and running "npm ci" or "npm install"
> on the destination platform after the copy. Or you could consider using yarn
> instead which has built-in support for installing a package on multiple
> platforms simultaneously.
> 
> If you are installing with yarn, you can try listing both this platform and the
> other platform in your ".yarnrc.yml" file using the "supportedArchitectures"
> feature: https://yarnpkg.com/configuration/yarnrc/#supportedArchitectures
> Keep in mind that this means multiple copies of esbuild will be present.
> 
> Another alternative is to use the "esbuild-wasm" package instead, which works
> the same way on all platforms. But it comes with a heavy performance cost and
> can sometimes be 10x slower than the "esbuild" package, so you may also not
> want to do that.
> 
> Error:
> You installed esbuild on another platform than the one you're currently using.
> This won't work because esbuild is written with native code and needs to
> install a platform-specific binary executable.
> 
> Specifically the "esbuild-darwin-arm64" package is present but this platform
> needs the "esbuild-darwin-64" package instead. People often get into this
> situation by installing esbuild on Windows or macOS and copying "node_modules"
> into a Docker image that runs Linux, or by copying "node_modules" between
> Windows and WSL environments.
> 
> If you are installing with npm, you can try not copying the "node_modules"
> directory when you copy the files over, and running "npm ci" or "npm install"
> on the destination platform after the copy. Or you could consider using yarn
> instead which has built-in support for installing a package on multiple
> platforms simultaneously.
> 
> If you are installing with yarn, you can try listing both this platform and the
> other platform in your ".yarnrc.yml" file using the "supportedArchitectures"
> feature: https://yarnpkg.com/configuration/yarnrc/#supportedArchitectures
> Keep in mind that this means multiple copies of esbuild will be present.
> 
> Another alternative is to use the "esbuild-wasm" package instead, which works
> the same way on all platforms. But it comes with a heavy performance cost and
> can sometimes be 10x slower than the "esbuild" package, so you may also not
> want to do that.
> 
>     at generateBinPath (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/esbuild/lib/main.js:1784:17)
>     at esbuildCommandAndArgs (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/esbuild/lib/main.js:1868:31)
>     at ensureServiceIsRunning (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/esbuild/lib/main.js:2030:25)
>     at Object.build (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/esbuild/lib/main.js:1923:26)
>     at bundleConfigFile (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/vite/dist/node/chunks/dep-9c153816.js:71364:34)
>     at loadConfigFromFile (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/vite/dist/node/chunks/dep-9c153816.js:71341:35)
>     at resolveConfig (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/vite/dist/node/chunks/dep-9c153816.js:70873:34)
>     at createServer (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/vite/dist/node/chunks/dep-9c153816.js:56425:26)
>     at start (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/@cypress/vite-dev-server/dist/startServer.js:64:36)
>     at processTicksAndRejections (node:internal/process/task_queues:96:5)
>     at async startDevServer (/Users/thiagosaiferodrigues/Documents/work/storyblok-js-client/node_modules/@cypress/vite-dev-server/dist/index.js:8:27)

I tried installing with npm and yarn. Bothe triggered this error. Any suggestion?

thiagosaife commented:
@alexjoverm Hello my friend. When you have the time, please take a look on my comment above. Thanks!

alexjoverm commented:
Hi @thiagosaife, sorry my friend it was a long time. Finally got back to this and updated all tests and configs from the latest `master`.

As per [this issue](https://github.com/evanw/esbuild/issues/2202), I've updated Node version to 16 (LTS), as cypress wasn't working on 14 and I believe it's safe to stay on latest LTS.

As a second note, as you can see cypress is running perfectly on the GH Action. If it gives you esbuild issues, please try to delete node_modules and `npm i` again. Let me know what you think ;)
